### PR TITLE
added svelte logic

### PR DESCRIPTION
**Pull request by oodtoon**
added lots of svelte logic.

### CODE REVIEW

File: src/routes/game-type/local/+page.svelte [hunk 1]
```diff
@@ -1,38 +1,209 @@
...
+  <MoveList />
+  <GameButtons />
+
+  <end-game-dialog class="end-game-dialog" />
+  <undo-dialog class="undo-dialog" />
+  <review-dialog class="review-dialog" />
+  <promotion-dialog class="promotion-dialog" />
 </div>
 
 <style>
```
Comment (dgore7) [inline src/routes/game-type/local/+page.svelte:41]: A lot of these styles can be purged

**Review by dgore7 — APPROVED** [review]

File: src/routes/+page.svelte [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <script>
     import boardImg from '$lib/images/board.png';
-    import "../components/ChessBoard.js"        
+    import "$lib/components/ChessBoard.js"
```
Comment (dgore7) [inline src/routes/+page.svelte]: This was deleted so this line breaks everything. I recommend removing

File: src/lib/store.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import type { Piece } from "$lib/models/pieces";
+import { writable, derived } from "svelte/store";
+
+export const selectedPiece = writable<Piece | null>();
+
+export const capturedPiece = writable<Piece>();
+export const capturedBlackPieces = writable<Piece[] | null[]>([]);
```
Comment (dgore7) [inline src/lib/store.ts]: Shouldn't this be Piece[]? I don't foresee this being null

File: src/lib/type.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export type Color = "White" | "Black";
```
Comment (dgore7) [inline src/lib/type.ts:1]: Capital letters - gross 🤮

File: src/lib/models/pieces/piece.ts [hunk 5]
```diff
@@ -1,18 +1,29 @@
...
-import Board from "../board.js";
-
-class Piece {
-  name = null;
+import Move from "../move";
+import Board from "../board";
+import type Game from "../game";
+import type Player from "../player";
+
+type Square = [number, number];
```
Comment (dgore7) [inline src/lib/models/pieces/piece.ts]: extracting this to a `types.ts` file might be a good idea

File: src/lib/type.ts (hunk 4 above)
Comment (oodtoon) [reply src/lib/type.ts:1]: too bad. It saves me from doing work for the dialogs.

File: src/lib/store.ts (hunk 3 above)
Comment (oodtoon) [reply src/lib/store.ts]: isn't it null prior to pieces being there?

**Pull request merged**